### ISSUE TITLE
"else if" breaks template includes

### ISSUE THREAD

**Issue by bendem** [fizzed/rocker#56]
The example below generates invalid code. Removing the `else if`, fixes it.

templates/test/Layout.rocker.html
```html
@args (String title, RockerBody content)

<!DOCTYPE html>
<html>
<head>
    <title>@title</title>
</head>
<body>

    @content

</body>
</html>
```

templates/test/Bug.rocker.html
```html
@args()

@templates.test.Layout.template("title") -> {
    @if (true) {
        hi
    } else if (true) {
        nope
    }
}
```

Generated code:
```java
package templates.test;

import java.io.IOException;
import com.fizzed.rocker.ForIterator;
import com.fizzed.rocker.RenderingException;
import com.fizzed.rocker.RockerContent;
import com.fizzed.rocker.RockerOutput;
import com.fizzed.rocker.runtime.DefaultRockerTemplate;
import com.fizzed.rocker.runtime.PlainTextUnloadedClassLoader;

/*
 * Auto generated code to render template templates/test/Bug.rocker.html
 * Do not edit this file. Changes will eventually be overwritten by Rocker parser!
 */
public class Bug extends com.fizzed.rocker.runtime.DefaultRockerModel {

    static public final com.fizzed.rocker.ContentType CONTENT_TYPE = com.fizzed.rocker.ContentType.HTML;
    static public final String TEMPLATE_NAME = "Bug.rocker.html";
    static public final String TEMPLATE_PACKAGE_NAME = "templates.test";
    static public final String HEADER_HASH = "69072368";
    static public final long MODIFIED_AT = 1491381164000L;
    static public final String[] ARGUMENT_NAMES = { };

    static public Bug template() {
        return new Bug();
    }

    @Override
    protected DefaultRockerTemplate buildTemplate() throws RenderingException {
        // optimized for performance (via rocker.optimize flag; no auto reloading)
        return new Template(this);
    }

    static public class Template extends com.fizzed.rocker.runtime.DefaultRockerTemplate {

        //         hi\n
        static private final byte[] PLAIN_TEXT_0_0;
        //         nope\n
        static private final byte[] PLAIN_TEXT_1_0;

        static {
            PlainTextUnloadedClassLoader loader = PlainTextUnloadedClassLoader.tryLoad(Bug.class.getClassLoader(), Bug.class.getName() + "$PlainText", "UTF-8");
            PLAIN_TEXT_0_0 = loader.tryGet("PLAIN_TEXT_0_0");
            PLAIN_TEXT_1_0 = loader.tryGet("PLAIN_TEXT_1_0");
        }

        public Template(Bug model) {
            super(model);
            __internal.setCharset("UTF-8");
            __internal.setContentType(CONTENT_TYPE);
            __internal.setTemplateName(TEMPLATE_NAME);
            __internal.setTemplatePackageName(TEMPLATE_PACKAGE_NAME);
        }

        @Override
        protected void __doRender() throws IOException, RenderingException {
            // ValueClosureBegin @ [3:1]
            __internal.aboutToExecutePosInTemplate(3, 1);
            __internal.renderValue(templates.test.Layout.template("title").__body(() -> {
                // IfBlockBegin @ [4:5]
                __internal.aboutToExecutePosInTemplate(4, 5);
                if (true) {
                    // PlainText @ [4:17]
                    __internal.aboutToExecutePosInTemplate(4, 17);
                    __internal.writeValue(PLAIN_TEXT_0_0);
                    // IfBlockElseIf @ [6:5]
                    __internal.aboutToExecutePosInTemplate(6, 5);
                } else if (true) {
                    // PlainText @ [6:23]
                    __internal.aboutToExecutePosInTemplate(6, 23);
                    __internal.writeValue(PLAIN_TEXT_1_0);
                    // IfBlockEnd @ [4:5]
                    __internal.aboutToExecutePosInTemplate(4, 5);
                } // if end @ [4:5]
                // ValueClosureEnd @ [3:1]
                __internal.aboutToExecutePosInTemplate(3, 1);
            } // value closure end @ [3:1]
        }
    }

    private static class PlainText {

        static private final String PLAIN_TEXT_0_0 = "        hi\n";
        static private final String PLAIN_TEXT_1_0 = "        nope\n";

    }

}
```

**Comment by jjlauer** [comment]
When you say "breaks" the template, do you mean the java compiler errors
out while compiling the Bug template?  In your example, the java compiler
probably does not like two "true" conditional statements -- so I'm just
curious if your example is valid.  Can you copy and paste what the compiler
is complaining about?

**Comment by bendem** [comment]
I mean, that the generated java is broken.

```java
__internal.renderValue(templates.test.Layout.template("title").__body(() -> {
    // ...
} // value closure end @ [3:1]
```
it's missing `);`

**Comment by jesusvimlet** [comment]
I'm experienced something similar.
When I use else if within a @with block, the compiled template has the following error:
} // with end @ [30:1]
while it should be:
}); // with end @ [30:1]

**Comment by jjlauer** [comment]
Fix for this issue is in master.  Will publish a new release soon.

**Comment by jjlauer** [comment]
Fixed in v0.17.0
